Source/UI/VirtusizeButton.swift

```diff
@@ -48,32 +50,19 @@ public class VirtusizeButton: UIButton, NewVirtusizeView {
 
 	required init?(coder aDecoder: NSCoder) {
 		super.init(coder: aDecoder)
+		virtusizeEventHandler = self
 		isHidden = true
-		addNotificationObserver()
 	}
 
 	public init() {
 		super.init(frame: .zero)
+		virtusizeEventHandler = self
 		isHidden = true
 		setStyle()
-		addNotificationObserver()
-	}
-
-	private func addNotificationObserver() {
-		NotificationCenter.default.addObserver(
-			self,
-			selector: #selector(validProductDataCheck(_:)),
-			name: .productDataCheck,
-			object: nil
-		)
 	}
-
-	@objc func validProductDataCheck(_ notification: Notification) {
-		let productWithPDC = notification.object as? VirtusizeProduct
-		if product?.externalId == productWithPDC?.externalId {
-			product = productWithPDC
-			isHidden = false
-		}
+	
+	deinit {
+		print("VirtusizeButton is dying!")
```

Comment:
😆

Reply:
I was testing something with `deinit` and forgot to remove it when I committed the code 😆